extensions/amp-render/1.0/amp-render.js

```diff
@@ -335,6 +336,35 @@ export class AmpRender extends BaseElement {
       .then((html) => dict({'__html': html}));
   }
 
+  /**
+   *
+   * @param {?Element} template
+   * @return {number} count of non-empty child nodes
+   * @private
+   */
+  getTemplateNonEmptyNodeCount_(template) {
+    let childNodes = [];
+    if (template.tagName === 'SCRIPT') {
+      const div = this.element.ownerDocument.createElement('div');
+      div./*OK*/ innerHTML = template./*OK*/ innerHTML;
+      const fragment = this.element.ownerDocument.createDocumentFragment();
+      fragment.append(div);
+      childNodes = fragment.firstElementChild.children;
```

Comment:
1. Why do we need the fragment? I believe you can get the child nodes from the created `div` directly.
2. `children` will [only contain elements](https://developer.mozilla.org/en-US/docs/Web/API/Element/children), but we need to take `TextNode`s into account as well, use `childNodes` instead.

Reply:
1. Was originally trying to use `DocumentFragment` but it's `append` method doesn't work with innerHTML strings. Anyways creating an element and appending to it is a better approach.
2. Good catch! Noted for the future.